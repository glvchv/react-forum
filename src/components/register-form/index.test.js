import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import RegisterForm from '.';

jest.mock('../error-message', () => 'ErrorMessage');

describe('Register-form component', () => {
    it('should render component', () => {
        const tree = renderer.create(
            <BrowserRouter>
                <RegisterForm />
            </BrowserRouter>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    })
})