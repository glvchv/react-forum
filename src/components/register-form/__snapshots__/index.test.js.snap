// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register-form component should render component 1`] = `
<div
  className="login-page"
>
  <div
    className="form"
  >
    <h1
      className="title"
    >
      Sign up
    </h1>
    <form
      className="login-form"
      onSubmit={[Function]}
    >
      <label
        className="label"
      >
        Username:
      </label>
      <input
        onChange={[Function]}
        placeholder="johndoe89"
        type="text"
      />
      <ErrorMessage
        error=""
      />
      <label
        className="label"
      >
        Password:
      </label>
      <input
        onChange={[Function]}
        placeholder="********"
        type="password"
      />
      <label
        className="label"
      >
        Repeat password:
      </label>
      <input
        onChange={[Function]}
        placeholder="********"
        type="password"
      />
      <ErrorMessage
        error=""
      />
      <button
        disabled={false}
        type="submit"
      >
        register
      </button>
      <p
        className="message"
      >
        Already registered? 
        <a
          href="/login"
          onClick={[Function]}
        >
          Sign in
        </a>
      </p>
    </form>
  </div>
</div>
`;
